app/uk/gov/hmrc/vatregisteredcompanies/repositories/PayloadBufferRepository.scala

```diff
@@ -16,61 +16,64 @@
 
 package uk.gov.hmrc.vatregisteredcompanies.repositories
 
+import org.bson.types.ObjectId
+
 import javax.inject.{Inject, Singleton}
-import play.api.Logger
+import play.api.Logging
 import play.api.libs.json._
-import play.modules.reactivemongo.ReactiveMongoComponent
-import reactivemongo.api.Cursor
-import reactivemongo.bson.{BSONDocument, BSONObjectID}
-import reactivemongo.play.json.ImplicitBSONHandlers._
-import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.play.json.formats.MongoFormats.Implicits.objectIdFormat
+import org.mongodb.scala.model.{Filters, Sorts}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 import uk.gov.hmrc.vatregisteredcompanies.models.Payload
 
 import scala.concurrent.{ExecutionContext, Future}
 
-final case class PayloadWrapper(
-  _id: BSONObjectID = BSONObjectID.generate,
+final case class PayloadWrapper (
+  _id: ObjectId = ObjectId.get(),
   payload: Payload
 )
 
 object PayloadWrapper {
-  val format: Format[PayloadWrapper] = Json.format[PayloadWrapper]
+  implicit  val format: Format[PayloadWrapper] = Json.format[PayloadWrapper]
 }
 
 @Singleton
 class   PayloadBufferRepository@Inject()(
-  reactiveMongoComponent: ReactiveMongoComponent
+  mongoComponent: MongoComponent
 )(implicit val executionContext: ExecutionContext)
-  extends ReactiveRepository(
-    "vatregisteredcompaniesbuffer",
-    reactiveMongoComponent.mongoConnector.db,
-    PayloadWrapper.format) {
+  extends PlayMongoRepository[PayloadWrapper](
+    mongoComponent = mongoComponent,
+    collectionName = "vatregisteredcompaniesbuffer",
+    domainFormat = PayloadWrapper.format,
+    indexes = Seq()) with Logging {
 
   def deleteAll(): Future[Unit] =
-    removeAll().map(_=> ())
-
-  implicit val format: OFormat[PayloadWrapper] = Json.format[PayloadWrapper]
+    collection.deleteMany(Filters.empty()).toFuture().map(_ => ())
 
   def insert(payload: Payload): Future[Unit] =
-    collection.insert(true).one(PayloadWrapper(payload = payload)).map(_ => ())
+    collection.insertOne(PayloadWrapper(payload = payload)).toFuture().map(_ => ())
 
   def list: Future[List[PayloadWrapper]] =
-    findAll()
+    collection.find(Filters.empty()).toFuture().map(_.toList)
 
   private def getOne: Future[List[PayloadWrapper]] = {
-    val query = BSONDocument()
     collection
-      .find(query, Option.empty[JsObject])
-      .sort(Json.obj("_id" -> 1))
-      .cursor[PayloadWrapper]()
-      .collect[List](1, Cursor.FailOnError[List[PayloadWrapper]]())
+      .find(Filters.empty())
+      .sort(Sorts.ascending("_id"))
+      .toFuture()
+      .map(_.toList)
   }
 
   def one: Future[Option[PayloadWrapper]] = getOne.map(_.headOption)
 
   def deleteOne(payload: PayloadWrapper): Future[Unit] = {
-    logger.info(s"deleting payload ${payload._id}")
-    remove("_id" -> payload._id).map { _ => (())}
+    collection.findOneAndDelete(Filters.equal("_id", payload._id))
```

Comment:
This could be a deleteOne rather than findAndDelete, we don't require the record to be returned so best to use the less computational heavy method.